### PR TITLE
chore: add update CMake workflow

### PR DESCRIPTION
**Pull request by mayeut**
towards https://github.com/scikit-build/cmake-python-distributions/discussions/186

### CODE REVIEW

File: .cmake.version [hunk 1]
```diff
@@ -0,0 +1 @@
+3.26.3
```
Comment (mayeut) [inline .cmake.version:1]: The added file is just there to ease getting a proper PR title / commit message.
For the commit message, bump -- commit could be used, requiring to setup the bot user before calling bump.

File: .github/workflows/update-cmake.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        nox --force-color -s bump
+        echo "version=$(cat .cmake.version)" >> $GITHUB_OUTPUT
+
+    # we use this step to grab a Github App auth token, so that PRs generated by this workflow
+    # run the GHA tests.
+    - uses: tibdex/github-app-token@v1
+      id: generate-token
+      if: github.ref == 'refs/heads/master' && github.repository == 'scikit-build/cmake-python-distributions'
+      with:
+        app_id: ${{ secrets.SCIKIT_BUILD_BOT_APP_ID }}
```
Comment (mayeut) [inline .github/workflows/update-cmake.yml:38]: a bot needs to be setup to get a proper token.

File: .github/workflows/update-cmake.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      with:
+        commit-message: Update dependencies
+        title: '[Bot] Update to CMake ${{ steps.bump-cmake.outputs.version }}'
+        body: |
+          Created by update_cmake_version.py
+
+          PR generated by "Update CMake" [workflow](https://github.com/${{github.repository}}/actions/runs/${{github.run_id}}).
+        branch: update-cmake-pr
+        committer: "scikit-build-bot[bot] <83877280+scikit-build-bot[bot]@users.noreply.github.com>"
+        author: "scikit-build-bot[bot] <83877280+scikit-build-bot[bot]@users.noreply.github.com>"
```
Comment (mayeut) [inline .github/workflows/update-cmake.yml:53]: this needs to be updated once a bot user actually exists.

File: .github/workflows/update-cmake.yml (hunk 2 above)
Comment (henryiii) [reply .github/workflows/update-cmake.yml:38]: Do you need this if you select the "create PRs" permission for GHA?

Comment (mayeut) [reply .github/workflows/update-cmake.yml:38]: from: https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#workarounds-to-trigger-further-workflow-runs
I understand that a PR could be created using the default GITHUB_TOKEN, however, this PR would not trigger a CI workflow run.

File: .github/workflows/update-cmake.yml (hunk 3 above)
Comment (jcfr) [reply .github/workflows/update-cmake.yml:53]: We have https://github.com/scikit-build-bot that could be used for that.

**Comment by henryiii** [comment]
This is still in draft.

File: .cmake.version (hunk 1 above)
Comment (henryiii) [reply .cmake.version:1]: This is in pyproject.toml now

File: .github/workflows/update-cmake.yml (hunk 3 above)
Comment (mayeut) [reply .github/workflows/update-cmake.yml]: I created an app at the organization level as users are less appropriate for this.

**Comment by mayeut** [comment]
sample PR in #512

File: .github/workflows/update-cmake.yml (hunk 2 above)
Comment (mayeut) [reply .github/workflows/update-cmake.yml]: I created an app at the organization level as users are less appropriate for this.

The app is installed only for this repo.
Secrets are set at organization level but limited to this repo for now.

**Pull request merged**